lib/date_time/date_time.c

```diff
@@ -656,29 +312,30 @@ int date_time_uptime_to_unix_time_ms(int64_t *uptime)
 int date_time_now(int64_t *unix_time_ms)
 {
 	int err;
-	int64_t unix_time_ms_prev;
+	struct timespec tp;
 
 	if (unix_time_ms == NULL) {
 		LOG_ERR("The passed in pointer cannot be NULL");
 		return -EINVAL;
 	}
+	if (!date_time_is_valid()) {
+		LOG_WRN("Valid time not currently available");
+		return -ENODATA;
+	}
 
-	unix_time_ms_prev = *unix_time_ms;
-
-	*unix_time_ms = k_uptime_get();
-
-	err = date_time_uptime_to_unix_time_ms(unix_time_ms);
+	err = clock_gettime(CLOCK_REALTIME, &tp);
```

Comment:
Most importantly, the test needs to run on `qemu_cortex_m3`. That's the only platform in CI library tests are run on, AFAIK. If that works, I think it's ok to remove support for `native_posix`.

Reply:
Could this be a configuration issue? That the prj.conf needs to be updated?

Reply:
Just did a few on qemu_cortex_m3 and it looks like clock_gettime() works as it should

Reply:
In the test _test_date_time_already_converted_ the test expects -EINVAL to be returned by `date_time_uptime_to_unix_time_ms`. But `date_time_valid` returns -ENODATA.

Reply:
I would run the test locally and try to sort out the test cases one by one. It's quite slow to depend on CI to build the test for you.

This is the alias I'm using to build for qemu_cortex_m3.
`alias mcq3='db && west build -b qemu_cortex_m3 && west build -t run'`

Reply:
Finally got QEMU working on my PC and starting to resolve qemu tests.

Reply:
Tests pass for QEMU.
`native_posix` doesn't work and based on above comments, I've removed `native_posix` support from `testcase.yaml`

Reply:
The problem with QEMU was that k_uptime_get() always returned zero in QEMU.

Googling helped and k_uptime_get() returns non-zero with CONFIG_QEMU_ICOUNT=n:
https://github.com/zephyrproject-rtos/zephyr/issues/26242

I'm not entirely sure this is perfect fix and that k_uptime_get() would return always non-zero because I'm seeing values of 10, 20 and 30 in the tests.

Reply:
I've added `initial_time_valid` back as `k_uptime_get()` returns zero in QEMU early in the tests so relying on it would be unstable and we don't want to debug those issues from failed CI runs.